hoomd/BoxDim.h

```diff
@@ -306,22 +306,22 @@ struct
 #ifdef __HIPCC__
         if (m_periodic.z)
             {
-            Scalar img = rint(w.z * m_Linv.z);
+            Scalar img = slow::rint(w.z * m_Linv.z);
```

Comment:
I know we talked about `rint` why are we using the slow version again?

Reply:
`slow` is misnamed here. Without qualifiers, `rint` calls the *double* precision function [`double rint(double)`](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html#group__CUDA__MATH__DOUBLE_1g3b8026edb2f2e441669845f0f3fa3bf7). In builds with `HOOMD_LONGREAL_TYPE=32`, this forces the delta r values to be converted to double, the `rint` called in double, then the result converted back to float. I made this change after I found this performance degradation while profiling.

`slow::rint` calls HOOMD's version of `rint` which uses template specialization to choose between `::rint` and [`::rintf`](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE_1g7791cd93108ffc6d24524f2e8635ccfd) (that works on floats). Both of these compile to a single SASS instruction and are fast. However, they are in `hoomd::math::slow` because they are *accurate*. The methods in `hoomd::math::fast` are [approximations on the GPU](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#intrinsic-functions) that are faster, but have reduced precision or are valid only for limited argument ranges.

Perhaps we should rename the hoomd math namespaces to `accurate` and `approximate`, but not in this pull request.

Reply:
That sounds good to me.